src/main/java/com/splunk/logtosplunk/loggable_events/LoggableBlockEvent.java

```diff
@@ -0,0 +1,146 @@
+package com.splunk.logtosplunk.loggable_events;
+
+import com.splunk.logtosplunk.actions.BlockEventAction;
+
+import net.minecraft.util.Vec3;
+
+/**
+ * Almost pojo with fields for information that might be associated with a block event.
+ */
+public class LoggableBlockEvent extends AbstractLoggableEvent {
+    private String playerName;
+    private String cause;
+    private BlockEventAction action;
+    private String blockName;
+    private String baseType;
+
+    /**
+     * Constructor.
+     *
+     * @param action The type of block action this represents, e.g. 'break'.
+     */
+    public LoggableBlockEvent(BlockEventAction action, long gameTime, String worldName, Vec3 location) {
+        super(LoggableEventType.BLOCK, gameTime, worldName, location);
+        this.action = action;
+    }
+
+    /**
+     * unused
+     */
+    public String getCause() {
+        return cause;
+    }
+
+    /**
+     * unused
+     */
+    public LoggableBlockEvent setCause(String cause) {
+        this.cause = cause;
+        return this;
+    }
+
+    public String getPlayerName() {
+        return playerName;
+    }
+
+    public LoggableBlockEvent setPlayerName(String playerName) {
+        this.playerName = playerName;
+        return this;
+    }
+
+    public BlockEventAction getAction() {
+        return action;
+    }
+
+    public String getBlockName() {
+        return blockName;
+    }
+
+    public LoggableBlockEvent setBlockName(String blockName) {
+        this.blockName = blockName;
+        return this;
+    }
+
+    public String getBaseType() {
+        return baseType;
+    }
+
+    public LoggableBlockEvent setBaseType(String baseType) {
+        this.baseType = baseType;
+        return this;
+    }
+
+    @Override
+    public String toString() {
+        StringBuilder b = new StringBuilder();
+        b.append(getType().getEventName()).append(": ").append(getAction().asString());
+        if (playerName != null) {
+            b.append(", By: ").append(playerName);
```

Comment:
the string pattern ", <key> <value>" repeats in the code a lot. Building a map and having a default formatter for this would be good. I'm thinking pass a map into the messagePreparer and have that format to the Splunk log style. If you want to keep it as flexible as a String you could still have a utility class that does this for you but a map should be flexible enough that if you have other messagePreparers it can fit the bill.